sway-ir/src/parser.rs

```diff
@@ -52,16 +52,34 @@ mod ir_builder {
                 }
 
             rule fn_decl() -> IrAstFnDecl
-                = "fn" _ name:id() _ selector:selector_id()? _ "(" _
-                      args:(block_arg() ** comma()) ")" _ "->" _ ret_type:ast_ty()
-                          metadata:comma_metadata_idx()? "{" _
-                      locals:fn_local()*
-                      blocks:block_decl()*
-                  "}" _ {
+                = "pub fn" _ name:id() _ selector:selector_id()? _ "(" _
```

Comment:
Yep, we want to remove `pub` and `mut` from the IR serialisation as with hindsight it's apparent that they aren't useful at this stage.  It's simpler to let the compiler worry about that stuff and to assume that all data are mutable and all symbols are public once we hit IR.

So I'd remove all these changes and instead go into `printer.rs` and fix the output of `function_to_doc` to not care about `pub`.